### PR TITLE
Tslint

### PR DESCRIPTION
**Pull request by xiao-lix**
- Enable tslint for each plugin, core, common, and AISKU & AISKULight
- Fix lint errors
- Edit tslint rules
- Add `rush lint`

### CODE REVIEW

File: shared/AppInsightsCommon/src/Util.ts [hunk 1]
```diff
@@ -60,17 +60,17 @@ export class Util {
      * @return {Storage} Returns storage object verified that it is usable
      */
     private static _getVerifiedStorageObject(storageType: StorageType): Storage {
-        var storage: Storage = null;
-        var fail: boolean;
-        var uid;
+        let storage: Storage = null;
+        let fail: boolean;
+        let uid;
```
Comment (markwolff) [inline shared/AppInsightsCommon/src/Util.ts]: change to `let uid: Date`

File: shared/AppInsightsCore/tsconfig.json [hunk 2]
```diff
@@ -11,7 +11,6 @@
     "noEmitHelpers": true,
     "alwaysStrict": true,
     "declaration": true,
-    "declarationMap": true,
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/tsconfig.json]: Why are you removing declarationMap?

File: shared/AppInsightsCore/package.json [hunk 3]
```diff
@@ -37,7 +38,8 @@
         "rollup-plugin-node-resolve": "^3.4.0",
         "rollup-plugin-replace": "^2.1.0",
         "rollup-plugin-uglify": "^6.0.0",
-        "tslint": "^5.11.0",
+        "tslint": "^5.19.0",
+        "tslint-config-prettier": "^1.18.0",
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/package.json]: What is this package doing?

File: shared/AppInsightsCommon/tslint.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (hectorhdzg) [inline shared/AppInsightsCommon/tslint.json]: Can you have a single tslint.json used by all packages?

File: shared/AppInsightsCommon/src/TelemetryItemCreator.ts [hunk 5]
```diff
@@ -32,21 +32,21 @@ export class TelemetryItemCreator {
...
+        const telemetryItem: ITelemetryItem = {
             name: envelopeName,
             time: new Date().toISOString(),
             iKey: "", // this will be set in TelemetryContext
             ext: systemProperties ? systemProperties : {}, // part A
             tags: [],
             data: {
             },
-            baseType: baseType,
+            baseType,
```
Comment (hectorhdzg) [inline shared/AppInsightsCommon/src/TelemetryItemCreator.ts]: Is this actually taking the parameter value?

File: shared/AppInsightsCore/package.json (hunk 3 above)
Comment (xiao-lix) [reply shared/AppInsightsCore/package.json]: it makes tslint and prettier work together without conflicts: https://www.npmjs.com/package/tslint-config-prettier

File: shared/AppInsightsCommon/tslint.json (hunk 4 above)
Comment (xiao-lix) [reply shared/AppInsightsCommon/tslint.json]: @markwolff I don't think so?

File: shared/AppInsightsCommon/src/Util.ts [hunk 6]
```diff
@@ -60,18 +60,18 @@ export class Util {
      * @return {Storage} Returns storage object verified that it is usable
      */
     private static _getVerifiedStorageObject(storageType: StorageType): Storage {
-        var storage: Storage = null;
-        var fail: boolean;
-        var uid;
+        let storage: Storage = null;
+        let fail: boolean;
+        let uid: Date;
```
[the remark this answers is not in the document]
Comment (xiao-lix) [reply shared/AppInsightsCommon/src/Util.ts]: @markwolff

File: shared/AppInsightsCommon/src/TelemetryItemCreator.ts (hunk 5 above)
Comment (xiao-lix) [reply shared/AppInsightsCommon/src/TelemetryItemCreator.ts]: yes, this is following [object-literal-shorthand rule](https://palantir.github.io/tslint/rules/object-literal-shorthand/) that enforce using [es6 object literal shorthand](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand).

File: shared/AppInsightsCommon/tslint.json (hunk 4 above)
Comment (markwolff) [reply shared/AppInsightsCommon/tslint.json]: You can have a `tslint.base.json` at the root dir which the tslint of each package would `"extends": "../../tslint.base.json"` on

**Pull request merged**